static/app/components/confirm.tsx

```diff
@@ -119,6 +119,10 @@ type Props = OpenConfirmOptions & {
    * Passed to `children` render function
    */
   disabled?: boolean;
+  /**
+   * It immediately invokes confirm modal after the component is mounted
+   */
+  isOpen?: boolean;
```

Comment:
Can you instead just call `openConfirmModal` in a useEffect? `isOpen` implies that it would be controlled, but setting it to false will not close the modal.

I would prefer if we didn't add this property I think.